UnitsNet/Extensions/QuantityExtensions.cs

```diff
@@ -0,0 +1,139 @@
+
+/// <summary>
+/// Provides extension methods applicable to all quantities in the UnitsNet library.
+/// </summary>
+public static class QuantityExtensions
+{
+    /// <inheritdoc cref="IQuantity.As(UnitKey)" />
+    /// <remarks>This should be using UnitConverter.Default.ConvertValue(quantity, toUnit) </remarks>
+    internal static double GetValue<TQuantity>(this TQuantity quantity, UnitKey toUnit)
+        where TQuantity : IQuantity
+    {
+        return quantity.As(toUnit);
+    }
+    
+    /// <summary>
+    ///     Returns the string representation of the specified quantity using the provided format provider.
+    /// </summary>
+    /// <typeparam name="TQuantity">
+    ///     The type of the quantity, which must implement <see cref="IQuantity" /> and
+    ///     <see cref="IFormattable" />.
+    /// </typeparam>
+    /// <param name="quantity">The quantity to convert to a string.</param>
+    /// <param name="formatProvider">
+    ///     The format provider to use for localization and number formatting.
+    ///     If <c>null</c>, the default is <see cref="CultureInfo.CurrentCulture" />.
+    /// </param>
+    /// <returns>A string representation of the quantity.</returns>
+    public static string ToString<TQuantity>(this TQuantity quantity, IFormatProvider? formatProvider)
+        where TQuantity : IQuantity, IFormattable
+    {
+        return quantity.ToString(null, formatProvider);
+    }
+
+    /// <summary>
+    ///     Returns the string representation of the specified quantity using the provided format string.
+    /// </summary>
+    /// <typeparam name="TQuantity">
+    ///     The type of the quantity, which must implement <see cref="IQuantity" /> and
+    ///     <see cref="IFormattable" />.
+    /// </typeparam>
+    /// <param name="quantity">The quantity to convert to a string.</param>
+    /// <param name="format">
+    ///     The format string to use for formatting the quantity.
+    ///     If <c>null</c> or empty, the default format is used.
+    /// </param>
+    /// <returns>A string representation of the quantity.</returns>
+    public static string ToString<TQuantity>(this TQuantity quantity, string? format)
+        where TQuantity : IQuantity, IFormattable
+    {
+        return quantity.ToString(format, null);
+    }
+
+    /// <summary>
+    ///     Calculates the arithmetic mean of a sequence of quantities.
+    /// </summary>
+    /// <typeparam name="TQuantity">The type of the quantity.</typeparam>
+    /// <param name="quantities">The collection of quantities to average.</param>
+    /// <returns>The average of the quantities, using the unit of the first element in the sequence.</returns>
+    /// <exception cref="ArgumentNullException">Thrown when the collection is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when the collection is empty.</exception>
+    internal static TQuantity ArithmeticMean<TQuantity>(this IEnumerable<TQuantity> quantities)
```

Comment:
There are a lot of mathy extension methods spread around, that seem related:
- QuantityExtensions.ArithmeticMean
- UnitMath
- GenericMathExtensions

I think we should gather these into a single file, but out of scope for this PR.

We can put everything here, in `QuantityExtensions` and a partial file, or in a separate `QuantityMathExtensions` class to isolate a bit for intellisense.

Reply:
Sounds good.

As long as `Sum()` and `Average()` still work for untyped `IQuantity`, without consumers having to know about linear vs affine vs logarithmic, I think we're good. 
If affine or logarithmic needs a different implementation or we can optimize in some way, we'll just delegate that internally.

Reply:
> And am pretty sure we couldn't have it both ways, as the extensions would become ambiguous.

Huh yeah, I hadn't thought of that.

I guess, the only way to support extensions on both untyped and typed quantity interfaces are:
- Different signatures: usually not viable
- Different namespaces: please god no
- Different names, like `SumUntyped`: 🤮 

I'd rather throw untyped `IQuantity` extensions under the bus to get typed extensions if there is no obvious workaround to support both.